api/src/database/dao.ts

```diff
@@ -0,0 +1,5 @@
+import Knex from 'knex';
```

Comment:
I'd probably create another folder under `src` called `projects` that contains the `ProjectDao` which is a class that takes the `dbClient` as a constructor argument (a.k.a. dependency injection).

**src/projects/ProjectDao.ts**
```
export class ProjectDao(private db: Knex = getDbClientInstance()) {
  async getAllProjects() {
    return dbClient.select().from('projects');
  }
}
```

Reply:
In that projects folder I'd also create another thing like `ProjectService` that injects the dao:
``` 
export class ProjectService(private dao: ProjectDao) {
  async getAllProjects() {
    return dao.getAllProjects()
  }
}
```

Then your routes would only use the `ProjectService` and only the `ProjectService` would use the `ProjectDao`.

Reply:
Creating that service layer gives you a place to put business logic. So your routes just bind http to the service, your dao just binds method calls to the database, and the service is where real logic would be contained. Also if you need to inject other things you could inject them to the service.